clientbuffer.cpp

```diff
@@ -134,15 +136,18 @@ class CClientBufferMod : public CModule
 void CClientBufferMod::OnAddClientCommand(const CString& line)
 {
     const CString identifier = line.Token(1);
+    const int timeLimit = line.Token(2).ToInt();
+
     if (identifier.empty()) {
-        PutModule("Usage: AddClient <identifier>");
+        PutModule("Usage: AddClient <identifier> [timelimit]");
         return;
     }
     if (HasClient(identifier)) {
         PutModule("Client already exists: " + identifier);
         return;
     }
-    AddClient(identifier);
+
+    AddClient(identifier, timeLimit > 0 ? timeLimit : m_iTimeLimit);
```

Comment:
This doesn't look right. If the user adds a client without specifying a time limit, and then changes the global time limit, the change will not be applied to the client.

I think it would be more correct to make `WithinTimeLimit` use the per-client time limit if it has been set by the user (is non-zero), and if not, use the global time limit.